xmtp_mls/src/storage/encrypted_store/group_intent.rs

```diff
@@ -67,6 +77,37 @@ pub struct StoredGroupIntent {
     pub publish_attempts: i32,
 }
 
+impl StoredGroupIntent {
+    /// Calculate the message id for this intent.
+    ///
+    ///
+    /// # Returns
+    /// Returns [`Option::None`] if [`StoredGroupIntent`] is not [`IntentKind::SendMessage`] or if
+    /// an error occurs during decoding of intent data for [`IntentKind::SendMessage`].
+    pub fn message_id(&self) -> Result<Option<Vec<u8>>, IntentError> {
```

Comment:
My only worry about this refactor is that people are going to use it _and_ decode the proto somewhere else separately to get at the other fields, so we're going to end up doing the same work twice.

`message_id` sounds like it would be a cheap little operation, but it's actually pretty expensive.

Reply:
Added a comment explaining how that is a bad idea, we should watch out for that but I also am not sure when we would access the contents of a PlainText envelope outside of publish success/failure. It could be worth storing the message_id separately on `StoredGroupIntent` as an `Option` which would avoid the problem altogether